com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/linking/LazyLinkingResource2.java

```diff
@@ -52,6 +53,28 @@
 // Node Model serialization is a restricted API of Xtext. Until Xtext provides a proper serialization API, we need to access its internal components.
 public class LazyLinkingResource2 extends DerivedStateAwareResource implements ILazyLinkingResource2 {
 
+  private static final long SLOW_INFERENCE_TIME_DEFAULT = TimeUnit.SECONDS.toMillis(15);
+  private static final String SLOW_INFERENCE_TIME_PROPERTY = "resourceloader.slowinferencetime"; //$NON-NLS-1$
+  private static final long SLOW_PARSE_TIME_DEFAULT = TimeUnit.SECONDS.toMillis(15);
+  private static final String SLOW_PARSE_TIME_PROPERTY = "resourceloader.slowparsetime"; //$NON-NLS-1$
+  private static long slowInferenceTime;
+  private static long slowParseTime;
+
+  static {
+    String slowInferenceTimeString = System.getProperty(SLOW_INFERENCE_TIME_PROPERTY);

```

Comment:
I suppose most of the time the property is not set, then the return value will be null and we go over the NumberFormatException. I'd rather do something like

 private static final int ONE_SECOND = 1000;
private static int slowInferenceTime= SLOW_INFERENCE_TIME_DEFAULT;

String slowInferenceTimeString = System.getProperty(SLOW_INFERENCE_TIME_PROPERTY);
    if (slowInferenceTimeString != null) {
      try {
        loopTimeout = Integer.parseInt(slowInferenceTimeString ) * ONE_SECOND;
      } catch (NumberFormatException e) {
        LOGGER.warn("Could not parse slow inference time setting: " + slowInferenceTimeString );
      }
    }

because otherwise if you made a typo in the configuration, you'll just quietly revert to default. This is hard to see. Also having exceptions on the standard execution path is not nice. Also it's easier to specify time in seconds when configuring the tool.